src/components/backend-ai-credential-view.ts

```diff
@@ -368,15 +402,12 @@ export default class BackendAICredentialView extends BackendAIPage {
   }
 
   /**
-   * Read the vfolder host information.
-   */
-  _readVFolderHostInfo() {
-    globalThis.backendaiclient.vfolder.list_hosts().then((response) => {
-      this.allowed_vfolder_hosts = response.allowed;
-      this.default_vfolder_host = response.default;
-      this.shadowRoot.querySelector('#allowed_vfolder-hosts').layout(true).then(()=>{
-        this.shadowRoot.querySelector('#allowed_vfolder-hosts').select(0);
-      });
+  * Get All Storage host information (superadmin-only)
+  */
+  _getAllStorageHostsInfo() {
+    globalThis.backendaiclient.vfolder.list_all_hosts().then((res) => {
```

Comment:
`list_all_hosts` is a superadmin-only API. Then, what happens when the user is a domain admin? (User page is exposed to domain admins, FYI.)

Reply:
(Domain) admin should be able to create, modify, and delete a keypair resource policy which is bound to a user who belongs to the same domain with the admin. If the create resource policy is not working in the admin's domain, I think it is a permission bug.

Reply:
@adrysn 
Sorry, I'm a bit confused about accessing resource policy according to your last comment. Since `admin` role users do have permission to "create", "modify", and "delete", they can see all storage hosts that are connected to manager so that they can add or remove them as "allowed storage hosts". Otherwise, we should only allow "Read" permission for `admin` role users. (For now, there's only one access point for listing up all storage host in WebUI, which is "creating/modifying (keypair) resource policy" dialog).

And also, I think It's better to handle this issue as a separate issue.

Reply:
The permission system for the domain user is a separate backend topic. In this PR, we will ignore this issue.